docs/source/frameworks/transformers.rst

```diff
@@ -1,38 +1,312 @@
+    @svc.api(input=Text(), output=JSON())
+    def unmask(input_series: str) -> list:
+        return runner.run(input_series)
 
-    # load the model back:
-    loaded = bentoml.transformers.load_model("text-generation-pipeline:latest")
+Pretrained Models
+-----------------
 
-    # Load a given model under `Runner` abstraction with `to_runner`
-    runner = bentoml.transformers.get(tag).to_runner()
-    runner.init_local()
-    batched_sentence = [
-        "I love you and I want to spend my whole life with you",
-        "I hate you, Lyon, you broke my heart.",
-    ]
-    runner.run(batched_sentence)
+Using pretrained models from the Hugging Face does not require saving the model first in the BentoML model store. A custom runner 
+can be implemented to download and run pretrained models at runtime.
 
+.. seealso::
+
+   See :ref:`Custom Runner <concepts/runner:Custom Runner>` to learn more.
+
+Serving a Pretrained Model
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+    :caption: service.py
+
+    import bentoml
+
+    from bentoml.io import Text, JSON
+    from transformers import pipeline
+
+    class PretrainedModelRunnable(bentoml.Runnable):
+        SUPPORTED_RESOURCES = ("cpu",)
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            self.unmasker = pipeline(task="fill-mask", model="distilbert-base-uncased")
+
+        @bentoml.Runnable.method(batchable=False)
+        def __call__(self, input_text):
+            return self.unmasker(input_text)
+
+    runner = bentoml.Runner(PretrainedModelRunnable, name="pretrained_unmasker")
+
+    svc = bentoml.Service('pretrained_unmasker_service', runners=[runner])
+
+    @svc.api(input=Text(), output=JSON())
+    def unmask(input_series: str) -> list:
+        return runner.run(input_series)
+
+Custom Pipelines
+----------------
+
+Transformers custom pipelines allow users to define their own pre and post-process logic and customize how input data is forwarded to 
+the model for inference.
+
+.. seealso::
+
+    `How to add a pipeline <https://huggingface.co/docs/transformers/main/en/add_new_pipeline>`_ from Hugging Face to learn more.
+
+.. code-block:: python
+    :caption: train.py
+    
+    from transformers import Pipeline
+
+    class MyClassificationPipeline(Pipeline):
+        def _sanitize_parameters(self, **kwargs):
+            preprocess_kwargs = {}
+            if "maybe_arg" in kwargs:
+                preprocess_kwargs["maybe_arg"] = kwargs["maybe_arg"]
+            return preprocess_kwargs, {}, {}
+
+        def preprocess(self, text, maybe_arg=2):
+            input_ids = self.tokenizer(text, return_tensors="pt")
+            return input_ids
+
+        def _forward(self, model_inputs):
+            outputs = self.model(**model_inputs)
+            return outputs
+
+        def postprocess(self, model_outputs):
+            return model_outputs["logits"].softmax(-1).numpy()
```

Comment:
cc @aarnphm : update to PairwiseClassification from transformers when new custom pipelines API is out.

Reply:
I wanted to add the new API but that will force the user to a newer version of `transformers`. Maybe we can update after a few months.

Reply:
We can provide a backward layer, don't worry to much about this for now